### ISSUE TITLE
Huge problems!

### ISSUE THREAD

**Issue by isabsent** [try2codesecure/ShellMS#21]
The result of the installing and a few attempts to run your code on **API 26 Android emulator under Android Studio 3.1.3** below:

F:\GitRepo\note_v2>adb install "E:\Common\Android_Apps\com.android.shellms_6.apk"
adb server is out of date.  killing...
* daemon started successfully *
1025 KB/s (381733 bytes in 0.363s)
Success

F:\GitRepo\note_v2>adb shell am startservice --user 0 -n com.android.shellms/.sendSMS -e contact +123456789 -e msg "Message"
adb server is out of date.  killing...
* daemon started successfully *
error: device unauthorized. Please check the confirmation dialog on your device.

F:\GitRepo\note_v2>adb shell am startservice --user 0 -n com.android.shellms/.sendSMS -e contact +123456789 -e msg "Message"
adb server is out of date.  killing...
* daemon started successfully *
error: device offline

F:\GitRepo\note_v2>adb shell am startservice --user 0 -n com.android.shellms/.sendSMS -e contact +123456789 -e msg "Message"
adb server is out of date.  killing...
* daemon started successfully *

F:\GitRepo\note_v2>adb shell am startservice --user 0 -n com.android.shellms/.sendSMS -e contact +123456789 -e msg "Message"
adb server is out of date.  killing...
ADB server didn't ACK
* failed to start daemon *
error:

F:\GitRepo\note_v2>adb shell am startservice --user 0 -n com.android.shellms/.sendSMS -e contact +123456789 -e msg "Message"
adb server is out of date.  killing...
* daemon started successfully *
Starting service: Intent { cmp=com.android.shellms/.sendSMS (has extras) }

F:\GitRepo\note_v2>adb shell am startservice --user 0 -n com.android.shellms/.sendSMS -e contact +123456789 -e msg "Message"
adb server is out of date.  killing...
* daemon started successfully *
error: device unauthorized. Please check the confirmation dialog on your device.

P.S. Permissions wasn't granted after installation and first run and should be enabled by hand directly in Android App settings,

**Comment by try2codesecure** [comment]
You're right. Permission handling is completely missing for API 23 and above. Should be added in next version. If you want, please contribute.

**Comment by isabsent** [comment]
No, it isn't a reason of problems. I have added all permissions by hand after installation your app on android emulator. Above log is the result of running your app with already granted permissions.
